File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

int count_letters(string text);
int count_words(string text);
int count_sentences(string text);

int main(void)
{
    //promt the user for a text
    string text = get_string("Text: ");

    //count the number of letters, words, sentences
    float letters_amount = count_letters(text);
    float words_amount = count_words(text);
    float sentences_amount = count_sentences(text);
    printf("%f - letters_F.\n", letters_amount);

    //%f words\n%f sentences\n
    //words_amount, sentences_amount

    //average number of letters per 100 words in text
    float l = (letters_amount / words_amount) * 100;
    //average number of sentences per 100 words in text
    float s = (sentences_amount / words_amount) * 100;

    //Coleman-Liau index
    float index = 0.0588 * l - 0.296 * s - 15.8;
    int indexNew = round(index);
    //printf ("INDEX %i\n", indexNew);


    if (indexNew >= 1 && indexNew < 16)
    {
        printf("Grade %i\n", indexNew);
    }
    else if (indexNew < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade 16+\n");
    }
}

//counting letters in text
int count_letters(string text)
{
    int count_letters = 0;

    for (int i = 0, n = strlen(text); i < n; i++)
    {
        int letterNumASCII = text[i];
        if ((letterNumASCII >= 65 && letterNumASCII <= 90) || (letterNumASCII >= 97 && letterNumASCII <= 122))
        {
            count_letters += 1;
        }
    }
    printf("%i - LETTERS_I.",count_letters);
    return count_letters;
}

//counting words in text
int count_words(string text)
{
    int count_words = 1;
    int space_ASCII = 32;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        if (text[i] == space_ASCII)
        {
            count_words += 1;
        }
    }

    return count_words;
}

//counting sentences in text
int count_sentences(string text)
{
    int count_sentences = 0;
    for (int i = 0, n = strlen(text); i < n; i++)
    {
        int letterNumASCII = text[i];
        // . || ? || !
        if (letterNumASCII == 33 || letterNumASCII == 46 || letterNumASCII == 63)
        {
            count_sentences += 1;
        }
    }
    return count_sentences;


}

File: sentimental-readability/readability.py
# TODO
from cs50 import get_string


def main():
    # prompt user for a text
    text = get_string("Text: ")

# find the amount of words
# any sequence of characters separated by spaces should count as a word
    words = 1
    for i in text:
        if i.isspace():
            words += 1
    #print(words, "words")


# find the amount of letters
# a letter is any lowercase character from a to z or any uppercase character from A to Z
    letters = 0
    for i in text:
        if i.isalpha():
            letters += 1
    #print(letters, "letters")

# find the amount of sentences
# any occurrence of a period, exclamation point, or question mark indicates the end of a sentence.
    sentences = 0
    for i in text:
        if i == "." or i == "!" or i == "?":
            sentences += 1
    #print(sentences, "sentences")


#  Coleman-Liau index is computed as 0.0588 * L - 0.296 * S - 15.8 and rounds to the nearest integer
# find L (letters_average) an average amount of letters per 100 words
    letters_average = (letters / words) * 100

# find S (sentences_average) the average number of sentences per 100 words in the text
    sentences_average = (sentences / words) * 100

# compute Coleman-Liau index
    index = 0.0588 * letters_average - 0.296 * sentences_average - 15.8
    index = round(index)
    #print(index, "index")

# print the result
    if index >= 16:
        print("Grade 16+")
    elif index < 1:
        print("Before Grade 1")
    else:
        print("Grade ", index)


main()
